clipping-service/app/app.py

```diff
@@ -155,93 +269,106 @@ def clip_app():
 
 @app.route("/clip", methods=['POST'])
 def clip():
-    #data = flask.request.json
-    #source_raster = data['cog_url']
-    #target_bbox = data['target_bbox']
-    #target_epsg = data['target_epsg']
-
-    #SOURCE_LOGGER.info(f"Clipping {source_raster} to {target_bbox}")
-
-    #for i in range(11):
-    #    time.sleep(1)
-    #    print(f"logging progress {i*10}%")
-    #    SOURCE_LOGGER.info(f"Progress: {i*10}%")
-    #SOURCE_LOGGER.info("Done!")
-
-    #return jsonify({
-    #    'status': 'success',
-    #    'message': 'Clipping finished'
-    #})
-
     parameters = request.get_json()
     app.logger.info(parameters)
 
-    if not parameters['cog_url'].startswith(TRUSTED_BUCKET):
-        app.logger.error("Invalid source raster, not in the known bucket: %s",
-                         parameters['cog_url'])
-        raise ValueError("Invalid COG provided.")
+    if not parameters['file_url'].startswith(TRUSTED_BUCKET):
+        app.logger.error("Invalid source file, not in the known bucket: %s",
+                         parameters['file_url'])
+        raise ValueError("Invalid source file provided.")
+
+    source_file_type = parameters['layer_type']
+    if source_file_type not in [RASTER, VECTOR]:
+        raise ValueError("Invalid file type.")
 
     target_bbox = parameters["target_bbox"]
 
-    # align the bounding box to a raster grid
-    source_raster_path = f'/vsicurl/{parameters["cog_url"]}'
-    source_raster_info = cached_raster_info(source_raster_path)
+    # align the bounding box
+    source_file_path = f'/vsicurl/{parameters["file_url"]}'
+    if source_file_type == VECTOR:
+        source_file_path = source_file_path.replace('.mvt', '.fgb')
```

Comment:
Are we assuming that the MVT is what we will be passing to the clipping layer?  If so, should we assert that it has the expected extension?